### DISCUSSION TITLE
Coturn IPv6 with External IP

### DISCUSSION THREAD

**Discussion by shyim** [coturn/coturn#1187] [Q&A]
Hey,

I am trying to get Coturn running with IPv4 and IPv6 together. What I did so far is in my config is:

```
use-auth-secret
static-auth-secret=xxxx
external-ip=138.2.0.0
external-ip=2603:c020:800e:2300:af23:0:0:0
```

I am unsure how the IPv6 address have to be formatted. With brackets `[xxxx]` or not. Without the rendered IP looks weird:
![image](https://user-images.githubusercontent.com/6224096/231595447-68d35c41-ec97-4340-8ef8-7a56ad6e3e54.png)

as it feels like it appends to the IPv6 address the actual Port. With brackets coturn seems to work, but it connects to IPv4 only.

Anyone maybe an idea how to get IPv6 properly running? 😅

**Comment by shyim** [comment]
I looked at the host machine for the udp ports and for some reason they listen only on the IPv4 network device. maybe that's the problem

**Reply by shyim** [reply]
`keep-address-family` seems to fix IPv6, but all users without IPV6 are now forced and get always an IPv6 address

**Comment by eakraly** [comment]
Hi @shyim !

You should see if `--external-ip=IPv6` has worked if you can see in logs that turnserver successfully attached to the interface.

**Reply by shyim** [reply]
For the "normal port" 3478 it is listing to IPv4/IPv6. But the "random" ports (I guess it's relay ports) when connect are only listing on IPv4. I got it working with `keep-address-family`, but then only IPV6 will be used
